src/modules/azure-azcopy-sync.ts

```diff
@@ -0,0 +1,134 @@
+import { Echo } from "./echo";
+import shell from "shelljs";
+import { CommandStringBuilder } from "../utilities/command-string-builder";
+import child_process from "child_process";
+
+// -----------------------------------------------------------------------------------------
+// #region Constants
+// -----------------------------------------------------------------------------------------
+
+const AZCOPY_COMMAND = "azcopy sync";
+
+// #endregion Constants
+
+// -----------------------------------------------------------------------------------------
+// #region Interfaces
+// -----------------------------------------------------------------------------------------
+
+interface BlobStorageUrlParts {
+    account: string;
+    container: string;
+    path?: string;
+    sasToken?: string;
+}
+
+interface SyncLocalFileOptions {
+    localFilePath: string;
+    destination: BlobStorageUrlParts;
+}
+
+interface SyncContainerOptions {
+    deleteDestination: boolean;
+    destination: BlobStorageUrlParts;
+    recursive: boolean;
+    source: BlobStorageUrlParts;
+}
+
+// #endregion Interfaces
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
```

Comment:
Consider splitting this up into `Private Functions` and `Public Functions`, with `Public Functions` (the `AzureAzcopySync` object) appearing first in the file - for someone coming into this file looking for a high-level picture of the exposed functionality, they'd want to see that first before the nitty gritty of the private functions